submissions/MosQuitO404/oop-exercise/index.js

```diff
@@ -0,0 +1,92 @@
+import { print } from './js/lib.js';
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/MosQuitO404/a-tiny-JS-world
+   Web app: https://mosquito404.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+   constructor(species, name, gender, legs, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.saying = saying;
+   }
+   
+   get showProperties() {
+      return ['species', 'name', 'gender', 'legs', 'saying'].map(property => this[property]).join('; ');
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor(species, name, gender, saying) {
+      super(species, name, gender, 4, saying);
+   }
+}
+
+class Dog extends Animal {
+   constructor(name, gender) {
+      super('Dog', name, gender, 'Woof-woof!');
+   }
+}
+
+class Cat extends Animal {
+   constructor(name, gender) {
+      super('Cat', name, gender, 'Meow-meow!');
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, hands, saying) {
+      super('Human', name, gender, 2, saying);
+      this.hands = hands;
+   }
+
+   get showProperties() {
+      return ['species', 'name', 'gender', 'legs', 'hands', 'saying'].map(property => this[property]).join('; ');
+   };
```

Comment:
> Look into method overloading and **calling inherited versions of methods via `super`** (original project docs offer links to relevant materials).

Take whatever base class creates and append values that this class owns.

This way child class implementation will not depend on base class.

Imagine we add a shared property that base class would handle. In current implementation you'd need to change code in two classes. OOP helps to avoid this and keep code DRY.

Reply:
@OleksiyRudenko Maybe like this?